src/fp6.rs

```diff
@@ -222,6 +364,85 @@ impl Fp6 {
             c2: t * c2,
         })
     }
+
+    /// Although this is labeled "vartime", it is only
+    /// variable time with respect to the exponent. It
+    /// is also not exposed in the public API.
+    pub fn pow_vartime(&self, by: &[u64]) -> Self {
+        // We use a 8-bit window.  A 7-bit window would use the least
+        // (weighed) number of squares and multiplications, but the code
+        // would be a bit trickier.  A smaller window (5- or 6-bit) might
+        // be even faster, as the lookup-table would fit in L1 cache.
+
+        // Precompute lut[i] = x^i for i in {0, ..., 255}
+        let mut lut : [Fp6; 256] = [Default::default(); 256];
+        lut[0] = Fp6::one();
+        lut[1] = *self;
+        for i in 1..128 {
+            lut[2*i] = lut[i].square();
+            lut[2*i + 1] = lut[2*i] * self;
+        }
+
+        let mut res = Fp6::one();
+        let mut first = true;
+        for j in (0..by.len()).rev() {
+            let e = by[j];
+            if first {
+                first = false;
+            } else {
+                for _ in 0..8 {
+                    res = res.square();
+                }
+            }
+
+            res *= lut[((e >> (7 * 8)) & 255u64) as usize];
+            for _ in 0..8 { res = res.square(); }
+            res *= lut[((e >> (6 * 8)) & 255u64) as usize];
+            for _ in 0..8 { res = res.square(); }
+            res *= lut[((e >> (5 * 8)) & 255u64) as usize];
+            for _ in 0..8 { res = res.square(); }
+            res *= lut[((e >> (4 * 8)) & 255u64) as usize];
+            for _ in 0..8 { res = res.square(); }
+            res *= lut[((e >> (3 * 8)) & 255u64) as usize];
+            for _ in 0..8 { res = res.square(); }
+            res *= lut[((e >> (2 * 8)) & 255u64) as usize];
+            for _ in 0..8 { res = res.square(); }
+            res *= lut[((e >> (1 * 8)) & 255u64) as usize];
+            for _ in 0..8 { res = res.square(); }
+            res *= lut[(e  & 255u64) as usize];
+        }
+        res
+    }
+
+    /// Attempts to convert a little-endian byte representation of
+    /// a scalar into an `Fp6`.
```

Comment:
I'm a little confused why "scalar" is mentioned here, and why it's little endian rather than big endian; encodings of e.g. Fp2 elements use big endian.

Reply:
It was analogous to the description of https://github.com/zkcrypto/bls12_381/blame/master/src/fp.rs#L165.